samples/VirtualMachines/Setup-Windows11VM-VHD.ps1

```diff
@@ -0,0 +1,2788 @@
+  
+  Creates VM with custom configuration.
+
+.EXAMPLE
+  .\Setup-Windows11VM-VHD.ps1 -VMName "PackagingVM" -ImageType "MSIXPackagingTool"
+  
+  Creates VM configured for MSIX packaging with MSIX Packaging Tool installed.
+
+.EXAMPLE
+  .\Setup-Windows11VM-VHD.ps1 -VMName "Windows11-VM" -UpdateScriptsOnly -ImageType "DevVM"
+  
+  Updates only the C:\Setup scripts in an existing VHD (fast testing).
+
+.NOTES
+  Requires:
+  - Administrator privileges
+  - Hyper-V enabled
+  - PowerShell 7+
+  - Windows 11 ISO file
+#>
+
+[CmdletBinding()]
+param(
+    [string]$VMName = "Windows11-VM",
+    [string]$VMPath = "$env:USERPROFILE\VMs",
+    [string]$IsoPath = "$env:USERPROFILE\Downloads\windows11.iso",
+    [uint64]$MemoryStartupBytes = 4GB,
+    [uint64]$VHDSizeBytes = 80GB,
+    [string]$SwitchName = $null,
+    [string]$AdminUser = "Admin",
+    [string]$AdminPassword = "Password123!",
+    [switch]$SkipDownload,
+    [string]$WinGetConfigPath = $null,
+    [ValidateSet("DevVM", "MSIXPackagingTool")]
+    [string]$ImageType = "DevVM",
+    [switch]$UpdateScriptsOnly,
+    [switch]$DebugLogging
+)
+
+#region Helper Functions
+
+function Write-DebugLog {
+    param([string]$Message)
+    if ($script:DebugLogging) {
+        Write-Host "    [DEBUG] $Message" -ForegroundColor DarkGray
+    }
+}
+
+function Ensure-Admin {
+    $currentUser = [Security.Principal.WindowsIdentity]::GetCurrent()
+    $principal = New-Object Security.Principal.WindowsPrincipal($currentUser)
+    $isAdmin = $principal.IsInRole([Security.Principal.WindowsBuiltInRole]::Administrator)
+    
+    if (-not $isAdmin) {
+        Write-Host "⚠️  This script requires administrator privileges." -ForegroundColor Yellow
+        Write-Host "   Re-launching as administrator..." -ForegroundColor Cyan
+        
+        $arguments = "-NoProfile -ExecutionPolicy Bypass -File `"$PSCommandPath`""
+        foreach ($param in $PSBoundParameters.GetEnumerator()) {
+            $arguments += " -$($param.Key)"
+            if ($param.Value -is [switch]) {
+                if ($param.Value) {
+                    # Switch is present, don't add value
+                }
+            } else {
+                $arguments += " `"$($param.Value)`""
+            }
+        }
+        
+        Start-Process pwsh -Verb RunAs -ArgumentList $arguments
+        exit
+    }
+}
+
+function Ensure-PowerShell7 {
+    if ($PSVersionTable.PSVersion.Major -lt 7) {
+        Write-Host "⚠️  PowerShell 7 or higher is required." -ForegroundColor Yellow
+        Write-Host "   Installing PowerShell 7 via WinGet..." -ForegroundColor Cyan
+        
+        winget install --id Microsoft.PowerShell --source winget --silent --accept-package-agreements --accept-source-agreements
+        
+        Write-Host "✓ PowerShell 7 installed. Please restart this script in PowerShell 7." -ForegroundColor Green
+        exit
+    }
+}
+
+function Enable-HyperV {
```

Comment:
What about SKUs where Hyper-V is not available (Home skus)? Will this still enable Hyper-V for them?

Reply:
added detection